community/cypher/frontend-3.2/src/test/scala/org/neo4j/cypher/internal/frontend/v3_2/ast/AddTest.scala

```diff
@@ -57,6 +57,25 @@ class AddTest extends InfixExpressionTestBase(Add(_, _)(DummyPosition(0))) {
 
     testValidTypes(CTNode, CTList(CTNode))(CTList(CTNode))
     testValidTypes(CTFloat, CTList(CTFloat))(CTList(CTFloat))
+
+    testValidTypes(CTList(CTAny), CTList(CTAny))(CTList(CTAny))
+  }
+
+  test("should handle covariant types") {
+    testValidTypes(CTString.covariant, CTString.covariant)(CTString)
+    testValidTypes(CTString.covariant, CTInteger)(CTString)
+    testValidTypes(CTString, CTInteger.covariant)(CTString)
+    testValidTypes(CTInteger.covariant, CTInteger.covariant)(CTInteger)
+    testValidTypes(CTInteger.covariant, CTFloat)(CTFloat)
+    testValidTypes(CTInteger, CTFloat.covariant)(CTFloat)
+    testValidTypes(CTFloat.covariant, CTFloat.covariant)(CTFloat.covariant)
```

Comment:
`CTFloat = CTFloat.covariant` so use `CTFloat` as expected here for consistency with the String test